src/test/java/com/connexta/ingest/IngestITests.java

```diff
@@ -14,36 +16,115 @@
+import org.springframework.core.io.Resource;
+import org.springframework.http.HttpEntity;
 import org.springframework.http.HttpMethod;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.MediaType;
+import org.springframework.http.client.MultipartBodyBuilder;
+import org.springframework.mock.http.client.MockClientHttpRequest;
 import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.util.JsonPathExpectationsHelper;
 import org.springframework.test.web.client.MockRestServiceServer;
-import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.reactive.server.WebTestClient;
+import org.springframework.util.MultiValueMap;
 import org.springframework.web.client.RestTemplate;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.wait.strategy.HttpWaitStrategy;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
 
-@SpringBootTest
+@SpringBootTest(webEnvironment = WebEnvironment.DEFINED_PORT)
 @ActiveProfiles("test")
-@AutoConfigureMockMvc
+@Testcontainers
+@AutoConfigureWebTestClient
 public class IngestITests {
 
   private static final byte[] TEST_FILE = "some-content".getBytes();
+  private static final String TEST_FILE_CONTENT_TYPE = "text/plain";
   private static final byte[] TEST_METACARD = "metacard-content".getBytes();
+  private static final Resource METACARD =
+      new ByteArrayResource(TEST_METACARD) {
+
+        @Override
+        public long contentLength() {
+          return TEST_METACARD.length;
+        }
+
+        @Override
+        public String getFilename() {
+          return "metacard.xml";
+        }
+      };
+  private static final String MINIO_ADMIN_ACCESS_KEY = "admin";
+  private static final String MINIO_ADMIN_SECRET_KEY = "12345678";
+  private static final int MINIO_PORT = 9000;
+
+  @Container
+  public static final GenericContainer minioContainer =
+      new GenericContainer("minio/minio:RELEASE.2019-07-10T00-34-56Z")
+          .withEnv("MINIO_ACCESS_KEY", MINIO_ADMIN_ACCESS_KEY)
+          .withEnv("MINIO_SECRET_KEY", MINIO_ADMIN_SECRET_KEY)
+          .withExposedPorts(MINIO_PORT)
+          .withCommand("server --compat /data")
+          .waitingFor(
+              new HttpWaitStrategy()
+                  .forPath("/minio/health/ready")
+                  .withStartupTimeout(Duration.ofSeconds(30)));
+
+  private static final MultiValueMap<String, HttpEntity<?>> TEST_INGEST_REQUEST_BODY;
+
+  static {
+    final MultipartBodyBuilder builder = new MultipartBodyBuilder();
+    builder.part(
+        "file",
+        new ByteArrayResource(TEST_FILE) {
+
+          @Override
+          public long contentLength() {
+            return TEST_FILE.length;
+          }
+
+          @Override
+          public String getFilename() {
+            return "originalFilename.txt";
+          }
+        });
```

Comment:
:pencil2: Define all test data in the same place, even it is is used once.
`private static final Resource FILE_RESOURCE = ...`